week7/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/data/PhotosInspirationRepository.kt

```diff
@@ -0,0 +1,38 @@
+package com.manuflowers.photoinspiration.data
+
+import androidx.lifecycle.LiveData
+import com.manuflowers.photoinspiration.data.local.database.PhotosDao
+import com.manuflowers.photoinspiration.data.local.preferences.PhotoInspirationPreferences
+import com.manuflowers.photoinspiration.data.models.PhotoEntity
+import com.manuflowers.photoinspiration.data.models.Success
+import com.manuflowers.photoinspiration.data.models.asPhotoEntity
+import com.manuflowers.photoinspiration.data.remote.networking.RemoteApiManager
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class PhotosInspirationRepository(
+    private val remoteApiManager: RemoteApiManager,
+    private val photosDao: PhotosDao,
+    private val preferences: PhotoInspirationPreferences
+) {
+
+    val isUserLoggedIn: Boolean
+        get() = preferences.userState
+
+    fun saveUserState(userState: Boolean) {
+        preferences.saveUserState(userState)
+    }
+
+    suspend fun fetchAndSavePhotos() {
+        withContext(Dispatchers.IO) {
```

Comment:
You don't need the withContext here, as the API/DB will do these for you, once you put the suspend modifiers there and use Retrofit/Room support out of the box!

Reply:
Done !